fsm.go

```diff
@@ -255,6 +257,9 @@ func (f *FSM) Cannot(event string) bool {
 // The last error should never occur in this situation and is a sign of an
 // internal bug.
 func (f *FSM) Event(event string, args ...interface{}) error {
+	f.eventMutex.Lock()
```

Comment:
I don't think we need channels. Let me try to explain better:

The intent here is to prevent illegal double transitions away from the same state. Something that makes no sense in a finite state machine.

The earlier changes we made were to enable callbacks to use the *FSM api from within a callback, including methods like Current() and Can()/Cannot(). But you could never use Event() from within a callback since that would be blocked by the "in transition" error. Unless, as I've found out now, your in the before_/_after(special case of same state src and dst) -event callbacks. Don't you agree that this is a bug?

And if it is, we should maybe find a solution where "in transition" is set before the before_-event callback is called, or we need to lock the Event()-method with another mutex.